roblox/client.py

```diff
@@ -550,3 +550,47 @@ def get_base_gamepass(self, gamepass_id: int) -> BaseGamePass:
         Returns: A BaseGamePass.
         """
         return BaseGamePass(client=self, gamepass_id=gamepass_id)
+
+    # Catalog
+    def get_catalog_items(self, catalog_item_array: List[TypedDict[catalog_id: int, catalog_item_type: Literal[1, 2]]]) -> List[CatalogItem]:
```

Comment:
See my first comment, that would address this TypedDict mess by allowing you to just pass BaseAssets or a new BaseBundle